### PR TITLE
doc: add release notes for #22729

### PR DESCRIPTION
**Pull request by vasild**
Add release notes for #22729.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by maflcko — COMMENTED** [review]
left some nits/questions

File: doc/release-notes-22729.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Notable changes
+===============
+
+P2P and network changes
+-----------------------
+
+- Previously, if Bitcoin Core would be listening for P2P connections, then
```
[the remark this answers is not in the document]
Comment (vasild) [reply doc/release-notes-22729.md]: Done.

File: doc/release-notes-22729.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  it would always bind on `127.0.0.1:8334` to listen for incoming Tor
+  connections. It was not possible to switch off this even if the node didn't
```
[the remark this answers is not in the document]
Comment (vasild) [reply doc/release-notes-22729.md]: Done.

File: doc/release-notes-22729.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  use Tor. For example, a configuration `bind=addr:port` would result in
```
[the remark this answers is not in the document]
Comment (vasild) [reply doc/release-notes-22729.md]: Applied the `s/For example/Previously/` change. Initially I wrote "for example `1.2.3.4:8333`" but changed it to "for example `addr:port`", now this covers all is not an example anymore.

**Comment by vasild** [comment]
`620bbc77d9...307659733e`: address suggestions

File: doc/release-notes-22729.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  it would always bind on `127.0.0.1:8334` to listen for incoming Tor
+  connections. It was not possible to switch this off even if the node didn't
+  use Tor. Previously, a configuration `bind=addr:port` would result in
+  binding on `addr:port` and on `127.0.0.1:8334`. This has been changed and now
+  `bind=addr:port` results in binding on `addr:port` only. The default behavior
+  of binding to `0.0.0.0:8333` and `127.0.0.1:8334` has not been changed.
+
+  If you are using `bind=...` without `bind=...=onion` and rely on the previous
+  behavior to accept incoming Tor connections at `127.0.0.1:8334`, you need to
+  make this explicit by using `bind=... bind=127.0.0.1:8334=onion`. (#22729)
```
[the remark this answers is not in the document]
Comment (vasild) [reply doc/release-notes-22729.md]: Done with some minor changes. Thanks!

**Comment by vasild** [comment]
`307659733e...a7f5d188cc`: address suggestions

**Comment by davidgumberg** [comment]
ACK https://github.com/bitcoin/bitcoin/commit/a7f5d188cc7a23b1946d48693ada22fe90d9ffe0

Nit: Documentation in [`doc/tor.md`](https://github.com/bitcoin/bitcoin/blob/9774a958b501a6d439a734e18b29e04f59f973f6/doc/tor.md?plain=1#L196) should also be updated:

```diff
 (obviously, replace the .onion address with your own). It should be noted that you still
 listen on all devices and another node could establish a clearnet connection, when knowing
 your address. To mitigate this, additionally bind the address of your Tor proxy:

-    ./bitcoind ... -bind=127.0.0.1
+    ./bitcoind ... -bind=127.0.0.1:8334=onion
```

**Comment by vasild** [comment]
`a7f5d188cc..6d33e13bd4`: append one commit with that update to `doc/tor.md`, thanks!

**Comment by davidgumberg** [comment]
reACK https://github.com/bitcoin/bitcoin/pull/30502/commits/6d33e13bd493ca6cee7b52b990e4822a28e35d0a

**Review by willcl-ark — APPROVED** [review]
ACK 6d33e13bd493ca6cee7b52b990e4822a28e35d0a

Release notes look good to me now, thanks!

**Pull request merged**